# Mouse gut and spine single cell transcriptomics

## Report #1

### Background

Mice were fed with control diet or supplemented with B-fructan. Spinal cord and cecum tissues were dissected, enzymatically digested, FACS-sorted and finally subjected to 10X single-cell library construction.

Libraries were sequenced on a NovaSeq Illumina instrument at an average depth of 500,000,000 150-cycles paired-end reads per library. 

Metadata of libraries is included here:


| scseq ID | Disease | Diet      | Tissues |	Mouse ID       |
|----------|---------|-----------|---------|-----------------|
| C1	     | MS	     | B-fructan | Gut	   | 7_BF-21-Cecum   |
| C2	     | MS	     | B-fructan | Gut	   | 8_BF-26-Cecum   |
| C3	     | MS	     | B-fructan | Gut	   | 9_BF-27-Cecum   |
| C4	     | MS	     | Control	 | Gut	   | 10_CON-23-Cecum |
| C5	     | MS	     | Control	 | Gut	   | 11_CON-25-Cecum |
| C6	     | MS	     | Control	 | Gut	   | 12_CON-29-Cecum |
| B7	     | MS	     | B-fructan | Spine	 | 1_CON-23-Spine  |
| B8	     | MS	     | B-fructan | Spine	 | 2_CON-25-Spine  |
| B9	     | MS	     | B-fructan | Spine	 | 3_CON-29-Spine  |
| B10	     | MS	     | Control	 | Spine	 | 4_BF-21-Spine   |
| B11	  	 | MS      | Control	 | Spine	 | 5_BF-26-Spine   |
| B12		   | MS      | Control	 | Spine	 | 6_BF-27-Spine   |

Libraries were quantified using the proprietary pipeline from 10X genomics cellranger. We decided to use this pipeline because it relies on genome and GTF files, which allows to use all annotated elements on the genome, including pseudo-genes and non-conding RNAs. Methods like kallisto bus or salmon alevin are based on only the annotated transcriptome and therefore only include protein-expressing genes.

The mouse genome was formatted for cellranger using script mkref_GRCm38.sh. Quantification of libraries was conducted with the following code:

```bash
# Path to the reference genome
REF_PATH="/home/dayhoff/sdd/juan/projects/heatherArmstrong/SC_study/mm10"

# Path to your FASTQ files
FASTQ_PATH="/home/dayhoff/sdd/juan/projects/heatherArmstrong/SC_study"

# Array of sample names
SAMPLES=("B10" "B11" "B12" "B7" "B8" "B9" "C1" "C2" "C3" "C4" "C5" "C6")

# Run cellranger count for each sample
for SAMPLE in "${SAMPLES[@]}"; do
	echo "Processing sample: ${SAMPLE}"
		        
	cellranger count \
	--id="${SAMPLE}_count" \
	--transcriptome="${REF_PATH}" \
	--fastqs="${FASTQ_PATH}" \
	--sample="${SAMPLE}" \
	--localcores=8 \
	--localmem=64 \
	> "${SAMPLE}_count.log" 2>&1
	echo "Finished processing sample: ${SAMPLE}"
done


```

