# Mouse gut and spine single cell transcriptomics

## Report #1

### Background

Mice were fed with control diet or supplemented with B-fructan. Spinal cord and cecum tissues were dissected, enzymatically digested, FACS-sorted and finally subjected to 10X single-cell library construction.

Libraries were sequenced on a NovaSeq Illumina instrument at an average depth of 500,000,000 150-cycles paired-end reads per library. 

Metadata of libraries is included here:


| scseq ID | Disease | Diet      | Tissues |	Mouse ID       |
|----------|---------|-----------|---------|-----------------|
| C1	     | MS	     | B-fructan | Gut	   | 7_BF-21-Cecum   |
| C2	     | MS	     | B-fructan | Gut	   | 8_BF-26-Cecum   |
| C3	     | MS	     | B-fructan | Gut	   | 9_BF-27-Cecum   |
| C4	     | MS	     | Control	 | Gut	   | 10_CON-23-Cecum |
| C5	     | MS	     | Control	 | Gut	   | 11_CON-25-Cecum |
| C6	     | MS	     | Control	 | Gut	   | 12_CON-29-Cecum |
| B7	     | MS	     | B-fructan | Spine	 | 1_CON-23-Spine  |
| B8	     | MS	     | B-fructan | Spine	 | 2_CON-25-Spine  |
| B9	     | MS	     | B-fructan | Spine	 | 3_CON-29-Spine  |
| B10	     | MS	     | Control	 | Spine	 | 4_BF-21-Spine   |
| B11	  	 | MS      | Control	 | Spine	 | 5_BF-26-Spine   |
| B12		   | MS      | Control	 | Spine	 | 6_BF-27-Spine   |


### Quantification

Libraries were quantified using the proprietary pipeline from 10X genomics cellranger. We decided to use this pipeline because it relies on genome and GTF files, which allows to use all annotated elements on the genome, including pseudo-genes and non-conding RNAs. Methods like kallisto bus or salmon alevin are based on only the annotated transcriptome and therefore only include protein-expressing genes.

The mouse genome was formatted for cellranger using script mkref_GRCm38.sh. Quantification of libraries was conducted with the following code:

```bash
# Path to the reference genome
REF_PATH="/home/dayhoff/sdd/juan/projects/heatherArmstrong/SC_study/mm10"

# Path to your FASTQ files
FASTQ_PATH="/home/dayhoff/sdd/juan/projects/heatherArmstrong/SC_study"

# Array of sample names
SAMPLES=("B10" "B11" "B12" "B7" "B8" "B9" "C1" "C2" "C3" "C4" "C5" "C6")

# Run cellranger count for each sample
for SAMPLE in "${SAMPLES[@]}"; do
	echo "Processing sample: ${SAMPLE}"
		        
	cellranger count \
	--id="${SAMPLE}_count" \
	--transcriptome="${REF_PATH}" \
	--fastqs="${FASTQ_PATH}" \
	--sample="${SAMPLE}" \
	--localcores=8 \
	--localmem=64 \
	> "${SAMPLE}_count.log" 2>&1
	echo "Finished processing sample: ${SAMPLE}"
done
```

### Quality control

Quantification matrices were then postprocessed with script postprocessing.R, which essentially uses Seurat.

In an ideal library, the number of mitochondrial reads should be low, lower than 5% and the correlation between mitochondrial RNA and total RNA (nCount_RNA) should be low, while the correlation between total RNA (nCount_RNA) and number of genes (n_Feature_RNA) should be high, ideally higher than 90%.

Thus, when filtering data, parameters should be chosen to adjust those correlation coefficients conveniently.

### Inspection of metrics: percentage of mt RNA, number of genes and total RNA

Based on initial inspection of metrics, it was decided to filter data according to the following thresholds:

##### Gut

Number of genes: nFeature_RNA > 200 & nFeature_RNA < 6000
Total RNA: nCount_RNA > 500 & nCount_RNA < 25000
Mitochondrial RNA %: 5%

##### Spine

Number of genes: nFeature_RNA > 200 & nFeature_RNA < 3500
Total RNA: nCount_RNA > 500 & nCount_RNA < 12000
Mitochondrial RNA %: 5%

Based on the plots after filtering, those thresholds seemed to be a good compromise.

#### Sample C1: Gut fructan
##### Before filtering
Metrics
<table>
  <tr>
    <td><img src="figures/C1_seurat_QCplot.png" alt="C1_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C1_seurat_feature-relationship.png" alt="C1_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/C1_seurat_QCplot_AF.png" alt="C2_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C1_seurat_feature-relationship_AF.png" alt="C1_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample C2: Gut fructan
##### Before filtering
Metrics
<table>
  <tr>
    <td><img src="figures/C2_seurat_QCplot.png" alt="C2_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C2_seurat_feature-relationship.png" alt="C2_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/C2_seurat_QCplot_AF.png" alt="C2_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C2_seurat_feature-relationship_AF.png" alt="C2_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample C3: Gut fructan
##### Before filtering
Metrics
<table>
  <tr>
    <td><img src="figures/C3_seurat_QCplot.png" alt="C3_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C3_seurat_feature-relationship.png" alt="C3_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/C3_seurat_QCplot_AF.png" alt="C3_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C3_seurat_feature-relationship_AF.png" alt="C3_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample C4: Gut control
##### Before filtering
Metrics
<table>
  <tr>
    <td><img src="figures/C4_seurat_QCplot.png" alt="C4_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C4_seurat_feature-relationship.png" alt="C4_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/C4_seurat_QCplot_AF.png" alt="C4_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C4_seurat_feature-relationship_AF.png" alt="C4_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample C5: Gut control
##### Before filtering
Metrics
<table>
  <tr>
    <td><img src="figures/C5_seurat_QCplot.png" alt="C5_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C5_seurat_feature-relationship.png" alt="C5_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/C5_seurat_QCplot_AF.png" alt="C5_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C5_seurat_feature-relationship_AF.png" alt="C5_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample C6: Gut control
##### Before filtering
Metrics
<table>
  <tr>
    <td><img src="figures/C6_seurat_QCplot.png" alt="C6_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C6_seurat_feature-relationship.png" alt="C6_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/C6_seurat_QCplot_AF.png" alt="C6_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/C6_seurat_feature-relationship_AF.png" alt="C6_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample B7: Spine fructan
##### Before filtering

Metrics

<table>
  <tr>
    <td><img src="figures/B7_seurat_QCplot.png" alt="B7_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B7_seurat_feature-relationship.png" alt="B7_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/B7_seurat_QCplot_AF.png" alt="B7_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B7_seurat_feature-relationship_AF.png" alt="B7_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample B8: Spine fructan

##### Before filtering

Metrics

<table>
  <tr>
    <td><img src="figures/B8_seurat_QCplot.png" alt="B8_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B8_seurat_feature-relationship.png" alt="B8_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/B8_seurat_QCplot_AF.png" alt="B8_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B8_seurat_feature-relationship_AF.png" alt="B8_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>


#### Sample B9: Spine fructan

##### Before filtering

Metrics

<table>
  <tr>
    <td><img src="figures/B9_seurat_QCplot.png" alt="B9_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B9_seurat_feature-relationship.png" alt="B9_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/B9_seurat_QCplot_AF.png" alt="B9_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B9_seurat_feature-relationship_AF.png" alt="B9_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample B10: Spine control

##### Before filtering

Metrics

<table>
  <tr>
    <td><img src="figures/B10_seurat_QCplot.png" alt="B10_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B10_seurat_feature-relationship.png" alt="B10_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/B10_seurat_QCplot_AF.png" alt="B10_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B10_seurat_feature-relationship_AF.png" alt="B10_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample B11: Spine control

##### Before filtering

Metrics

<table>
  <tr>
    <td><img src="figures/B11_seurat_QCplot.png" alt="B11_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B11_seurat_feature-relationship.png" alt="B11_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/B11_seurat_QCplot_AF.png" alt="B11_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B11_seurat_feature-relationship_AF.png" alt="B11_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

#### Sample B12: Spine control

##### Before filtering

Metrics

<table>
  <tr>
    <td><img src="figures/B12_seurat_QCplot.png" alt="B12_seurat_QCplot.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B12_seurat_feature-relationship.png" alt="B12_seurat_feature-relationship.png" width="500"></td>
  </tr>
</table>

##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/B12_seurat_QCplot_AF.png" alt="B12_seurat_QCplot_AF.png" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/B12_seurat_feature-relationship_AF.png" alt="B12_seurat_feature-relationship_AF.png" width="500"></td>
  </tr>
</table>

### Inspection and characterization of individual datasets

Datasets were first imported individually and subjected to preprocessing and UMAP plotting:

#### Sample C1: Gut fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/C1_UMAPplot_seurat_clusters.png" alt="C1_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  C1  before filtering had  639  cells <br>
Sample  C1  after filtering had  542  cells

#### Sample C2: Gut fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/C2_UMAPplot_seurat_clusters.png" alt="C2_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  C2  before filtering had  3133  cells <br>
Sample  C2  after filtering had  2539  cells

#### Sample C3: Gut fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/C3_UMAPplot_seurat_clusters.png" alt="C3_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  C3  before filtering had  2842  cells <br>
Sample  C3  after filtering had  2561  cells 

#### Sample C4: Gut control

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/C4_UMAPplot_seurat_clusters.png" alt="C4_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  C4  before filtering had  2738  cells <br>
Sample  C4  after filtering had  2129  cells 

#### Sample C5: Gut control

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/C5_UMAPplot_seurat_clusters.png" alt="C5_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  C5  before filtering had  2747  cells <br>
Sample  C5  after filtering had  2308  cells 

#### Sample C6: Gut control

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/C6_UMAPplot_seurat_clusters.png" alt="C6_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  C6  before filtering had  2279  cells <br>
Sample  C6  after filtering had  1906  cells 

#### Sample B7: Spine fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/B7_UMAPplot_seurat_clusters.png" alt="B7_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  B7  before filtering had  1547  cells <br>
Sample  B7  after filtering had  1420  cells 

#### Sample B8: Spine fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/B8_UMAPplot_seurat_clusters.png" alt="B8_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  B8  before filtering had  1192  cells <br>
Sample  B8  after filtering had  1029  cells 

#### Sample B8: Spine fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/B9_UMAPplot_seurat_clusters.png" alt="B9_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  B9  before filtering had  1574  cells <br>
Sample  B9  after filtering had  1450  cells

#### Sample B10: Spine fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/B10_UMAPplot_seurat_clusters.png" alt="B10_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  B10  before filtering had  2918  cells <br>
Sample  B10  after filtering had  2717  cells 

#### Sample B11: Spine fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/B11_UMAPplot_seurat_clusters.png" alt="B11_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  B11  before filtering had  3376  cells <br>
Sample  B11  after filtering had  2935  cells

#### Sample B12: Spine fructan

##### UMAP plot

<table>
  <tr>
    <td><img src="figures/B12_UMAPplot_seurat_clusters.png" alt="B12_UMAPplot_seurat_clusters.png" width="500"></td>
  </tr>
</table>

##### cells
Sample  B12  before filtering had  3152  cells <br>
Sample  B12  after filtering had  2930  cells 

### Merging datasets

#### Gut datasets

All gut and all spine samples were merged into a single dataset, but keeping the original annotations. UMAP plots were generated and were coloured by:

##### 1. Original identity

Here we observe that sample C6 control (12_CON-29-Cecum) behaves differently from the other two in the same gorup (C4 and C5). My suggestion is to eliminate this sample, but it is ultimately up to you.

<table>
  <tr>
    <td><img src="figures/merged_datasets_gut_UMAP_by_orig_ident.png" alt="merged_datasets_gut_UMAP_by_orig_ident.png" width="500"></td>
  </tr>
</table>

##### 2. By diet

Here we observe, again the distribution of sample C6 deviated from the general distribution.

<table>
  <tr>
    <td><img src="figures/merged_datasets_gut_UMAP_by_Diet.png" alt="merged_datasets_gut_UMAP_by_Diet.png" width="500"></td>
  </tr>
</table>

##### 3. Seurat clusters

Each cluster here likely represent a different type of cell, there are two ways to label them, either by identifying a profile of the same tissue in mouse in the published literature, or investigating, also in the literature, which markers are available for the types of cells expeted in the cecum.

<table>
  <tr>
    <td><img src="figures/merged_datasets_gut_UMAP_by_seurat_clusters.png" alt="merged_datasets_gut_UMAP_by_seurat_clusters.png" width="500"></td>
  </tr>
</table>

Some cell types expected in cecum are:

##### Intestinal epithelial cells
Enterocytes
Goblet cells
Paneth cells
Enteroendocrine cells
Transit amplifying cells
Intestinal stem cells
Absorptive enterocytes

##### Immune cells such as
Macrophages
Dendritic cells
Resident macrophages
T cells
B cells
Innate lymphoid cells

##### Stromal cells
Fibroblasts
Myofibroblasts
Endothelial cells

##### Mesenchymal Cells
Smooth muscle cells


#### Spine datasets

All spine and all spine samples were merged into a single dataset, but keeping the original annotations. UMAP plots were generated and were coloured by:

##### 1. Original identity

Unlike gut samples, all samples from spinal cord appear to behave as expected. So, no samples are suggested to be removed.

<table>
  <tr>
    <td><img src="figures/merged_datasets_spine_UMAP_by_orig_ident.png" alt="merged_datasets_spine_UMAP_by_orig_ident.png" width="500"></td>
  </tr>
</table>

##### 2. By diet

Again, everything looks good.

<table>
  <tr>
    <td><img src="figures/merged_datasets_spine_UMAP_by_Diet.png" alt="merged_datasets_gut_UMAP_by_Diet.png" width="500"></td>
  </tr>
</table>

##### 3. Seurat clusters

Each cluster here likely represent a different type of cell, there are two ways to label them, either by identifying a profile of the same tissue in mouse in the published literature, or investigating, also in the literature, which markers are available for the types of cells expeted in the spine.

<table>
  <tr>
    <td><img src="figures/merged_datasets_gut_UMAP_by_Diet.png" alt="merged_datasets_gut_UMAP_by_Diet.png" width="500"></td>
  </tr>
</table>

Some expected types of cells in spinal cord are: 

##### Neuronal cells
Motor neurons
Interneurons
Sensory neuron projections

##### Glial cells:
Astrocytes
Oligodendrocytes
Microglia
NG2 cells/Oligodendrocyte precursor cells

##### Progenitor/Stem cells
Neural progenitor cells

##### Endothelial and Perivascular Cells
##### Ependymal cells
##### Pericytes
##### Meningeal cells

Cell type classification still needs to be done.

### Identification of markers per cluster

Preliminary automated marker exploration produced some relatively clear markers for spine clusters, but way less informative for cecum cells.

#### Putative markers for gut (cecum) cells

##### Heatmap

<table>
  <tr>
    <td><img src="figures/gut_merged_heatmap_top5_markers.png" alt="gut_merged_heatmap_top5_markers.png" width="500"></td>
  </tr>
</table>

##### Dotplot

<table>
  <tr>
    <td><img src="figures/gut_merged_dotplot_top5_markers.png" alt="gut_merged_dotplot_top5_markers.png" width="500"></td>
  </tr>
</table>


#### Putative markers for spinal cord cells

##### Heatmap

<table>
  <tr>
    <td><img src="figures/spine_merged_heatmap_top5_markers.png" alt="gut_merged_heatmap_top5_markers.png" width="500"></td>
  </tr>
</table>

##### Dotplot

<table>
  <tr>
    <td><img src="figures/spine_merged_dotplot_top5_markers.png" alt="gut_merged_dotplot_top5_markers.png" width="500"></td>
  </tr>
</table>


### Differential expression analysis

For each tissue, differential expression analysis, between control and B-fructan groups was conducted using the Wilcoxon test. 

#### Differentially expressed genes for gut (cecum) cells

<table>
  <tr>
    <td><img src="figures/gut2_EnhancedVolcano_B-fructan_vs_Control.png" alt="gut_EnhancedVolcano_B-fructan_vs_Control.png" width="500"></td>
  </tr>
</table>

Please see file gut_DE_results_Wilcoxon_Control_vs_B-fructan.xlsx

#### Differentially expressed genes for spinal cord cells

<table>
  <tr>
    <td><img src="figures/spine_EnhancedVolcano_B-fructan_vs_Control.png" alt="spine_EnhancedVolcano_B-fructan_vs_Control.png" width="500"></td>
  </tr>
</table>

Please see file spine_DE_results_Wilcoxon_Control_vs_B-fructan.xlsx