# Mouse gut and spine single cell transcriptomics

## Report #2

### Background

NOTE: In a previous iteration, data was analyzed in R, using the Seurat pipeline. Here, because some models for cell classification in the mouse gut (for cecum cells classification) and some mouse brain that could serve at least as guides to classify spinal cord cells.

Mice were fed with control diet or supplemented with B-fructan. Spinal cord and cecum tissues were dissected, enzymatically digested, FACS-sorted and finally subjected to 10X single-cell library construction.

Libraries were sequenced on a NovaSeq Illumina instrument at an average depth of 500,000,000 150-cycles paired-end reads per library. 

Metadata of libraries is included here:


| scseq ID | Disease | Diet      | Tissues |	Mouse ID       |
|----------|---------|-----------|---------|-----------------|
| C1	     | MS	     | B-fructan | Gut	   | 7_BF-21-Cecum   |
| C2	     | MS	     | B-fructan | Gut	   | 8_BF-26-Cecum   |
| C3	     | MS	     | B-fructan | Gut	   | 9_BF-27-Cecum   |
| C4	     | MS	     | Control	 | Gut	   | 10_CON-23-Cecum |
| C5	     | MS	     | Control	 | Gut	   | 11_CON-25-Cecum |
| C6	     | MS	     | Control	 | Gut	   | 12_CON-29-Cecum |
| B7	     | MS	     | B-fructan | Spine	 | 1_CON-23-Spine  |
| B8	     | MS	     | B-fructan | Spine	 | 2_CON-25-Spine  |
| B9	     | MS	     | B-fructan | Spine	 | 3_CON-29-Spine  |
| B10	     | MS	     | Control	 | Spine	 | 4_BF-21-Spine   |
| B11	  	 | MS      | Control	 | Spine	 | 5_BF-26-Spine   |
| B12		   | MS      | Control	 | Spine	 | 6_BF-27-Spine   |


### Quantification

Libraries were quantified using the proprietary pipeline from 10X genomics cellranger. We decided to use this pipeline because it relies on genome and GTF files, which allows to use all annotated elements on the genome, including pseudo-genes and non-conding RNAs. Methods like kallisto bus or salmon alevin are based on only the annotated transcriptome and therefore only include protein-expressing genes.

The mouse genome was formatted for cellranger using script mkref_GRCm38.sh. Quantification of libraries was conducted with the following code:

```bash
# Path to the reference genome
REF_PATH="/home/dayhoff/sdd/juan/projects/heatherArmstrong/SC_study/mm10"

# Path to your FASTQ files
FASTQ_PATH="/home/dayhoff/sdd/juan/projects/heatherArmstrong/SC_study"

# Array of sample names
SAMPLES=("B10" "B11" "B12" "B7" "B8" "B9" "C1" "C2" "C3" "C4" "C5" "C6")

# Run cellranger count for each sample
for SAMPLE in "${SAMPLES[@]}"; do
	echo "Processing sample: ${SAMPLE}"
		        
	cellranger count \
	--id="${SAMPLE}_count" \
	--transcriptome="${REF_PATH}" \
	--fastqs="${FASTQ_PATH}" \
	--sample="${SAMPLE}" \
	--localcores=8 \
	--localmem=64 \
	> "${SAMPLE}_count.log" 2>&1
	echo "Finished processing sample: ${SAMPLE}"
done
```

### Quality control

Quantification matrices were then postprocessed using a (postProcessing_mouseSC.ipynb) iPython workbook and the pipeline Scanpy.

In an ideal library, the number of mitochondrial reads should be low, lower than 5% and the correlation between mitochondrial RNA and total RNA (nCount_RNA) should be low, while the correlation between total RNA (nCount_RNA) and number of genes (n_Feature_RNA) should be high, ideally higher than 90%.

Thus, when filtering data, parameters should be chosen to adjust those correlation coefficients conveniently.

### Inspection of metrics: percentage of mt RNA, number of genes and total RNA

Based on initial inspection of metrics, it was decided to filter data according to the following thresholds:

##### Gut

# Filter cells based on QC metrics
cecum_adata_combined = cecum_adata_combined[
    (cecum_adata_combined.obs["pct_counts_mt"] < 10) &
    (cecum_adata_combined.obs["n_genes_by_counts"] < 6000) &
    (cecum_adata_combined.obs["n_genes_by_counts"] > 200)
]

##### Spine

spine_adata_combined = spine_adata_combined[
    (spine_adata_combined.obs["pct_counts_mt"] < 10) &
    (spine_adata_combined.obs["n_genes_by_counts"] < 6000) &
    (spine_adata_combined.obs["n_genes_by_counts"] > 200)
]
Based on the plots after filtering, those thresholds seemed to be a good compromise.

All cecum and spines were merged and labeled according to the metadata table shown above.

##### Tissue: Cecum
##### Before filtering
Metrics
<table>
  <tr>
    <td><img src="figures/violin_cecum_merged_plot.png" alt="Cecum samples before filtering" width="500"></td>
  </tr>
</table>

Correlation
<table>
  <tr>
    <td><img src="figures/scatter_cecum_merged_plot.png" alt="Relation between number of genes and total RNA" width="500"></td>
  </tr>
</table>

##### Tissue: Spine
##### Before filtering
Metrics
<table>
  <tr>
    <td><img src="figures/violin_spine_merged_plot.png" alt="Spine samples before filtering" width="500"></td>
  </tr>
</table>

Correlation
<table>
  <tr>
    <td><img src="figures/scatter_spine_merged_plot.png" alt="Relation between number of genes and total RNA" width="500"></td>
  </tr>
</table>

##### Tissue: Cecum
##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/violin_cecum_merged_plot_afterFiltering.png" alt="Cecum samples after filtering" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/scatter_cecum_merged_plot_afterFiltering.png" alt="Relation between number of genes and total RNA after filtering" width="500"></td>
  </tr>
</table>

##### Tissue: Spine
##### After filtering

Metrics
table>
  <tr>
    <td><img src="figures/violin_spine_merged_plot_afterFiltering.png" alt="Spine samples after filtering" width="500"></td>
  </tr>
</table>

Correlations
<table>
  <tr>
    <td><img src="figures/scatter_spine_merged_plotafterFiltering.png" alt="Relation between number of genes and total RNA after filtering" width="500"></td>
  </tr>
</table>


### Inspection and characterization of individual samples in merged objects

Merged datasets were subjected to preprocessing and UMAP plotting, colouring the plots by sample:

##### Tissue: Cecum

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_plot.png" alt="Cecum tissue UMAP colored by sample" width="500"></td>
  </tr>
</table>

As can be seen in the plots above, cecum samples overlap pretty good. Except an orange cluster on the left (sample C2).
  
##### Tissue: Spine

<table>
  <tr>
    <td><img src="figures/umap_spine_merged_UMAP_plot.png" alt="Spine tissue UMAP colored by sample" width="500"></td>
  </tr>
</table>

As can be seen in the plots above, spine samples overlap pretty good. However, a cluster, enclosed in a square looks interesting and deserves atention as it could be a fructan-induced subpopulation of cells.


### Inspection and characterization of groups in merged objects

Merged datasets were subjected to preprocessing and UMAP plotting, colouring the plots by group:

##### Tissue: Cecum

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_plot_byDiet.png" alt="Cecum tissue UMAP colored by diet" width="500"></td>
  </tr>
</table>

As can be seen in the plots above, cecum groups overlap pretty good. Except a blue cluster on the left (which corresponds to sample C2 and it is likely not group-specific, but sample specific).
  
##### Tissue: Spine

<table>
  <tr>
    <td><img src="figures/umap_spine_merged_UMAP_plot_byDiet.png" alt="Spine tissue UMAP colored by diet" width="500"></td>
  </tr>
</table>

As can be seen in the plots above, spine samples overlap pretty good. However, a cluster in blue, corresponding to the fructan group, is interesting and deserves atention as it could be a fructan-induced subpopulation of cells.


### Inspection of quality metrics by Leiden clusters

#### Diagnostic and removal of doublets

Doublets correspond to droplets with more than a single cell. Scanpy offers the Scrublet method to identify and subsequently remove doublets. It was applied, and this is how the data looks after doublets removal.

##### Tissue: Cecum

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_doublets_plot.png" alt="Cecum tissue UMAP colored by abundance of doublets" width="750"></td>
  </tr>
</table>


##### Tissue: Spine

<table>
  <tr>
    <td><img src="figures/umap_spine_merged_UMAP_doublets_plot.png" alt="Spine tissue UMAP colored by abundance of doublets" width="750"></td>
  </tr>
</table>


#### Canonical QC metrics per Leiden cluster

As mentioned above, number of genes, total RNA and percentage of mitochondrial RNA are typical QC metrics to inspect. Here they are presented on a UMAP projection.


##### Tissue: Cecum

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_QC_plots.png" alt="Cecum tissue UMAP colored by QC metrics" width="750"></td>
  </tr>
</table>

##### Tissue: Spine

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_QC_plots.png" alt="Spine tissue UMAP colored by QC metrics" width="750"></td>
  </tr>
</table>


#### Cell classification

For cell classification, CellTypist was used. See reference [here](https://pubmed.ncbi.nlm.nih.gov/38134877/). CellTypist contains a series of very useful models, listed here after:

|CellTypist models                             |
|----------------------------------------------|
| 01. Adult_COVID19_PBMC.pkl                   |
| 02. Adult_CynomolgusMacaque_Hippocampus.pkl  |
| 03. Adult_Human_MTG.pkl                      |
| 04. Adult_Human_PancreaticIslet.pkl          |
| 05. Adult_Human_PrefrontalCortex.pkl         |
| 06. Adult_Human_Skin.pkl                     |
| 07. Adult_Human_Vascular.pkl                 |
| 08. Adult_Mouse_Gut.pkl                      |
| 09. Adult_Mouse_OlfactoryBulb.pkl            |
| 10. Adult_Pig_Hippocampus.pkl                |
| 11. Adult_RhesusMacaque_Hippocampus.pkl      |
| 12. Autopsy_COVID19_Lung.pkl                 |
| 13. COVID19_HumanChallenge_Blood.pkl         |
| 14. COVID19_Immune_Landscape.pkl             |
| 15. Cells_Adult_Breast.pkl                   |
| 16. Cells_Fetal_Lung.pkl                     |
| 17. Cells_Human_Tonsil.pkl                   |
| 18. Cells_Intestinal_Tract.pkl               |
| 19. Cells_Lung_Airway.pkl                    |
| 20. Developing_Human_Brain.pkl               |
| 21. Developing_Human_Gonads.pkl              |
| 22. Developing_Human_Hippocampus.pkl         |
| 23. Developing_Human_Organs.pkl              |
| 24. Developing_Human_Thymus.pkl              |
| 25. Developing_Mouse_Brain.pkl               |
| 26. Developing_Mouse_Hippocampus.pkl         |
| 27. Fetal_Human_AdrenalGlands.pkl            |
| 28. Fetal_Human_Pancreas.pkl                 |
| 29. Fetal_Human_Pituitary.pkl                |
| 30. Fetal_Human_Retina.pkl                   |
| 31. Fetal_Human_Skin.pkl                     |
| 32. Healthy_Adult_Heart.pkl                  |
| 33. Healthy_COVID19_PBMC.pkl                 |
| 34. Healthy_Human_Liver.pkl                  |
| 35. Healthy_Mouse_Liver.pkl                  |
| 36. Human_AdultAged_Hippocampus.pkl          |
| 37. Human_Colorectal_Cancer.pkl              |
| 38. Human_Developmental_Retina.pkl           |
| 39. Human_Embryonic_YolkSac.pkl              |
| 40. Human_IPF_Lung.pkl                       |
| 41. Human_Longitudinal_Hippocampus.pkl       |
| 42. Human_Lung_Atlas.pkl                     |
| 43. Human_PF_Lung.pkl                        |
| 44. Human_Placenta_Decidua.pkl               |
| 45. Immune_All_High.pkl                      |
| 46. Immune_All_Low.pkl                       |
| 47. Lethal_COVID19_Lung.pkl                  |
| 48. Mouse_Dentate_Gyrus.pkl                  |
| 49. Mouse_Isocortex_Hippocampus.pkl          |
| 50. Mouse_Postnatal_DentateGyrus.pkl         |
| 51. Mouse_Whole_Brain.pkl                    |
| 52. Nuclei_Lung_Airway.pkl                   |
| 53. Pan_Fetal_Human.pkl                      |


For our cecum cells, the model 'Adult_Mouse_Gut.pkl' seems appropriate. However, there is no model for spinal cord. Initially, the SeqSeek model, originally published in the paper A harmonized atlas of mouse spinal cord cell types and their spatial organization (Nature Communications volume 12, Article number: 5722 (2021)), link [here](https://www.nature.com/articles/s41467-021-25125-1) wanted to be used, but unfortunately could not be installed successfully. This pipeline still deserves interest, but perhaps needs to be reimplemented, or the authors be contacted for help with installation.

Because of that issue, the model 'Mouse_Whole_Brain.pkl' was used to conduct an initial classification of spine cells.

Hereafter, UMAP plots are presented by tissue, colored by Leiden cluster, by predicted label and by voting majority. CellTypist classify cells according to two different criteria. A very granular scheme (name predicted_labels) classify cells by the type of individual cells. Another, more coarse, scheme, named majority voting is also implemented. Here, a cluster is colored and classify according to the majority of cells in such a clusters. Both approaches are complementary.

#### Results of cells classification

Both cecum and spine cells were classified using CellTypist as described above.


##### Tissue: Cecum

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_QC_plots.png" alt="Cecum tissue UMAP colored by QC metrics" width="750"></td>
  </tr>
</table>

#### Canonical QC metrics per Leiden cluster

As mentioned above, number of genes, total RNA and percentage of mitochondrial RNA are typical QC metrics to inspect. Here they are presented on a UMAP projection.


##### Tissue: Cecum

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_QC_plots.png" alt="Cecum tissue UMAP colored by QC metrics" width="750"></td>
  </tr>
</table>

##### Tissue: Spine

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_QC_plots.png" alt="Spine tissue UMAP colored by QC metrics" width="750"></td>
  </tr>
</table>

#### Canonical QC metrics per Leiden cluster

As mentioned above, number of genes, total RNA and percentage of mitochondrial RNA are typical QC metrics to inspect. Here they are presented on a UMAP projection.


##### Tissue: Cecum

Colored by Leiden clusters.

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_afterClustering_plot.png" alt="Cecum tissue UMAP colored by Leiden's clusters" width="750"></td>
  </tr>
</table>

Colored by predicted labels.

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_by-predicted_label_plot.png" alt="Cecum tissue UMAP colored by CellTypist predicted labels" width="750"></td>
  </tr>
</table>

Colored by majority voting.

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_by-majority_voting_plot_edited.png" alt="Cecum tissue UMAP colored by CellTypist majority voting" width="750"></td>
  </tr>
</table>

NOTE: This is the nicest cell classification I have seen. It is granular enough and very discrete.


##### Tissue: Spine

Colored by Leiden clusters.

<table>
  <tr>
    <td><img src="figures/umap_cecum_merged_UMAP_afterClustering_plot.png" alt="Spine tissue UMAP colored by Leiden's clusters" width="750"></td>
  </tr>
</table>

Colored by predicted labels.

<table>
  <tr>
    <td><img src="figures/umap_spine_merged_UMAP_by-predicted_label_plot.png" alt="Spine tissue UMAP colored by CellTypist predicted labels" width="750"></td>
  </tr>
</table>

Colored by majority voting.

<table>
  <tr>
    <td><img src="figures/umap_spine_merged_UMAP_by-majority_voting_plot_edited.png" alt="Spine tissue UMAP colored by CellTypist majority voting" width="750"></td>
  </tr>
</table>


NOTE: Because no specific model for spinal cord is available, this is only an approximation using mouse brain cells as reference. Some improvement needed are detection of astrocytes, epndymal, and Swann cells, which are subpopulations of glial cells. Neurons, Pericytes (vascular cells) and Fibrobasts (mesenchymal cells).

#### Differential expression analysis  

For each tissue, differential expression analysis, between control and B-fructan groups was conducted using the Wilcoxon test.

#### Differentially expressed genes for gut (cecum) cells

<table> 
  <tr>
    <td><img src="figures/cecum_volcano_plot_with_labels.png" alt="Volcano plot, DE genes cecum" width="500"></td>  
  </tr>    
</table>   

Please see file cecum_differential_expression.xlsx

#### Differentially expressed genes for spinal cord cells

<table>    
  <tr>     
    <td><img src="figures/spine_volcano_plot_with_labels.png" alt="Volcano plot, DE genes spline" width="500"></td>
  </tr>
</table>

Please see file spine_differential_expression.xlsx.


#### Gene ontology analysis

Gene ontology analysis was conducted with [ClusterProfiler](https://www.sciencedirect.com/science/article/pii/S2666675821000667).

##### Cecum

Upregulated genes

<table>
  <tr>     
    <td><img src="figures/cecum_geneOntology_UpregulatedGenes_dotplot.png" alt="Dotplot cecum gene ontology upregulated" width="500"></td>
  </tr>
</table>

Downregulated genes

<table>
  <tr>
    <td><img src="figures/cecum_geneOntology_DownregulatedGenes_dotplot.png" alt="Dotplot cecum gene ontology downregulated" width="500"></td>
  </tr>
</table>


Please see files cecum_GO_results_upregulated.xlsx and cecum_GO_results_downregulated.xlsx for a complete list of enriched or depleted GO terms in cecum.

##### Spine

Upregulated genes

<table>
  <tr>     
    <td><img src="figures/spine_geneOntology_UpregulatedGenes_dotplot.png" alt="Dotplot spine gene ontology upregulated" width="500"></td>
  </tr>
</table>

Downregulated genes

<table>
  <tr>
    <td><img src="figures/spine_geneOntology_DownregulatedGenes_dotplot.png" alt="Dotplot spine gene ontology downregulated" width="500"></td>
  </tr>
</table>

Please see files spine_GO_results_upregulated.xlsx and spine_GO_results_downregulated.xlsx for a complete list of enriched or depleted GO terms in spine.

